---
permalink: /download/
title: Download
layout: base
version: v4
---

{% assign version = page.version %} {% assign genome =
site.data.versions[version].genome %} {% assign tissues =
site.data.versions[version].tissues %}

<p>
  Download pages: <a href="/download/v1/">v1</a> •
  <a href="/download/v2/">v2</a> • <a href="/download/v3/">v3</a> •
  <a href="/download/v3-unmerged/">v3 unmerged</a> • v4 •
  <a href="/download/study-data/">Study data</a>
</p>

<h1>Download data</h1>

<p>
  See the <a href="/about/#data-specs">About page</a> for processing, data
  format, and RatGTEx version number specifications. File names include the
  version number, and files that include genomic coordinates also include the
  genome build for clarity, even though only one genome build is used for each
  version.
</p>

<p><a href="/data/{{ version }}/gene.{{ version }}.tsv">Gene info</a></p>

<h2>Gene expression</h2>

<p>
  <a href="/data/{{ version }}/expr/medianGeneExpression.{{ version }}.tsv.gz">
    Median TPM per gene per tissue
  </a>
  <em style="color: gray">Used for heatmap visualizations</em>
</p>

<strong>log2(readcount+1)</strong>
<em style="color: gray">Used to compute allelic fold change</em>
<p>
  {% for tissue in tissues %}
  <a
    href="/data/{{ version }}/expr/expr.log2.{{ tissue.id }}.{{ version }}_{{ genome }}.bed.gz"
    title="{{ tissue.name }}"
  >
    {{ tissue.id }}
  </a>
  {% unless forloop.last %} • {% endunless %} {% endfor %}
</p>

<strong>TPM (transcripts per million)</strong>
<em style="color: gray">Relative abundance of transcripts</em>
<p>
  {% for tissue in tissues %}
  <a
    href="/data/{{ version }}/expr/expr.tpm.{{ tissue.id }}.{{ version }}_{{ genome }}.bed.gz"
    title="{{ tissue.name }}"
  >
    {{ tissue.id }}
  </a>
  {% unless forloop.last %} • {% endunless %} {% endfor %}
</p>

<br />

<h2>All RNA phenotypes</h2>

<p>
  The TPM files above are the same as the unnormalized expression files below.
</p>

<strong>Unnormalized</strong>
<em style="color: gray">TPM, ratios, etc.</em>
<table class="download-matrix">
  <tbody>
    {% assign modalities = site.data.modalities %} {% for modality in modalities
    %}
    <tr>
      <td>{{ modality.name }}</td>
      {% for tissue in tissues %}
      <td>
        <a
          href="/data/{{ version }}/phenos/phenos.{{ tissue.id }}.{{ modality.id }}.unnorm.{{ version }}_{{ genome }}.bed.gz"
          title="{{ tissue.name }}"
        >
          {{ tissue.id }}
        </a>
      </td>
      {% endfor %}
    </tr>
    {% endfor %}
  </tbody>
</table>

<br />

<strong>Normalized</strong>
<em style="color: gray">Quantile-normalized and inverse-normal-transformed</em>
<table class="download-matrix">
  <tbody>
    {% assign modalities = site.data.modalities %} {% for modality in modalities
    %}
    <tr>
      <td>{{ modality.name }}</td>
      {% for tissue in tissues %}
      <td>
        <a
          href="/data/{{ version }}/phenos/phenos.{{ tissue.id }}.{{ modality.id }}.norm.{{ version }}_{{ genome }}.bed.gz"
          title="{{ tissue.name }}"
        >
          {{ tissue.id }}
        </a>
      </td>
      {% endfor %}
    </tr>
    {% endfor %}
  </tbody>
</table>

<br />

<h2>eQTLs</h2>

<p>
  <strong>
    <a
      href="/data/{{ version }}/eqtl/eqtls_indep.{{ version }}_{{ genome }}.tsv"
    >
      Conditionally independent cis-eQTLs
    </a>
  </strong>
  <br />
  A table of cis-eQTLs from all tissues.
</p>

<p>
  <strong>
    <a href="/data/{{ version }}/eqtl/top_assoc.{{ version }}_{{ genome }}.tsv">
      Top association per gene
    </a>
  </strong>
  <br />
  A table of the strongest variant-gene association per gene per tissue, even if
  not significant. A q-value threshold of 0.05 was used to determine significant
  eQTLs.
</p>

<strong>trans-eQTLs</strong>
<br />
All SNP-gene associations genome-wide with TSS distance &gt; 5 Mb and with an
arbitrary cutoff of p &lt; 1e-5. No significance testing was performed.
<p>
  {% for tissue in tissues %}
  <a
    href="/data/{{ version }}/xqtl/trans_qtl_pairs.{{ tissue.id }}.expression.{{ version }}_{{ genome }}.txt.gz"
    title="{{ tissue.name }}"
  >
    {{ tissue.id }}
  </a>
  {% unless forloop.last %} • {% endunless %} {% endfor %}
</p>

<br />

<h2>All xQTLs</h2>

<p>
  <strong>Conditionally independent cis-xQTLs for all tissues</strong>
  <br />
  <a
    href="/data/{{ version }}/xqtl/xqtls_indep.cross_modality.{{ version }}_{{ genome }}.tsv"
  >
    Cross-modality mapped</a
  >
  <em style="color: gray">
    (conditionally independent across modalities per gene per tissue)
  </em>
  <br />
  Each modality mapped separately: {% for modality in site.data.modalities %}
  <a
    href="/data/{{ version }}/xqtl/xqtls_indep.{{ modality.id }}.{{ version }}_{{ genome }}.tsv"
    title="{{ modality.name }}"
  >
    {{ modality.id }}
  </a>
  {% unless forloop.last %} • {% endunless %} {% endfor %}
</p>

<strong>All significant cis-xQTL variant-phenotype pairs</strong>
<table class="download-matrix">
  <tbody>
    {% for modality in site.data.modalities %}
    <tr>
      <td>{{ modality.name }}</td>
      {% for tissue in tissues %}
      <td>
        <a
          href="/data/{{ version }}/xqtl/cis_qtl_signif.{{ tissue.id }}.{{ modality.id }}.{{ version }}_{{ genome }}.txt.gz"
          title="{{ tissue.name }}"
        >
          {{ tissue.id }}
        </a>
      </td>
      {% endfor %}
    </tr>
    {% endfor %}
  </tbody>
</table>

<br />

<p>
  <strong>Top association per gene for all tissues</strong>
  <br />
  A table of the strongest variant-gene association per gene per tissue, even if
  not significant. A q-value threshold of 0.05 was used to determine significant
  xQTLs.
  <br />
  <a
    href="/data/{{ version }}/xqtl/top_assoc.cross_modality.{{ version }}_{{ genome }}.tsv"
  >
    Cross-modality mapped
  </a>
  <br />
  Each modality mapped separately: {% for modality in site.data.modalities %}
  <a
    href="/data/{{ version }}/xqtl/top_assoc.{{ modality.id }}.{{ version }}_{{ genome }}.tsv"
    title="{{ modality.name }}"
  >
    {{ modality.id }}
  </a>
  {% unless forloop.last %} • {% endunless %} {% endfor %}
</p>

<br />

<h2>Other info</h2>

<p>
  <a href="/data/{{ version }}/geno/ratgtex_v4_round11_2.{{ genome }}.vcf.gz">
    Genotypes
  </a>
</p>

<p>
  <a href="/data/{{ version }}/ref/RatGTEx_rats.{{ version }}.tsv"
    >Rat info (all tissues)</a
  >
</p>

<p>
  <a href="/data/{{ version }}/ref/RatGTEx_samples.{{ version }}.tsv">
    Sample info (all tissues)
  </a>
</p>

<strong>Rat IDs</strong>
<em style="color: gray">
  The final QC-filtered sample sets used for eQTL mapping
</em>
<br />
<p>
  {% for tissue in tissues %}
  <a
    href="/data/{{ version }}/rat_ids/rat_ids.{{ tissue.id }}.{{ version }}.txt"
    title="{{ tissue.name }}"
  >
    {{ tissue.id }}
  </a>
  {% unless forloop.last %} • {% endunless %} {% endfor %}
</p>

<strong>RNA-Seq FASTQ file lists</strong>
<em style="color: gray">with corresponding rat IDs</em>
<br />
Each line includes one FASTQ path for single-end reads, or two FASTQ paths for
paired-end reads, followed by the rat ID. FASTQ maps can include a mix of
single-end and paired-end rows. These files may include mappings to samples that
were filtered out during QC, so use the rat ID lists above to get the final
sample sets.
<br />
<p>
  {% for tissue in tissues %}
  <a
    href="/data/{{ version }}/fastq_map/fastq_map.{{ tissue.id }}.{{ version }}.txt"
    title="{{ tissue.name }}"
  >
    {{ tissue.id }}
  </a>
  {% unless forloop.last %} • {% endunless %} {% endfor %}
</p>

<strong>Covariates</strong>
<em style="color: gray">used for xQTL mapping</em>
<table class="download-matrix">
  <tbody>
    {% for modality in site.data.modalities %}
    <tr>
      <td>{{ modality.name }}</td>
      {% for tissue in tissues %}
      <td>
        <a
          href="/data/{{ version }}/covar/covar.{{ tissue.id }}.{{ modality.id }}.{{ version }}.tsv"
          title="{{ tissue.name }}"
        >
          {{ tissue.id }}
        </a>
      </td>
      {% endfor %}
    </tr>
    {% endfor %}
  </tbody>
</table>

<br />

<h2>API</h2>

<p>
  Different types of data used for visualizations, including expression, eQTLs,
  and reference annotations, can be accessed via API. See the
  <a href="/about/api/">API documentation</a> for details.
</p>

<br />

<h2>Raw data</h2>

<ul>
  <li>
    RNA-Seq reads for the IL, LHb, NAcc, OFC, and PL brain regions are available
    in SRA and can be accessed from GEO accession
    <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE173141">
      GSE173141</a
    >.
  </li>
  <li>
    RNA-Seq reads for Eye are available at GEO accession
    <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE201236">
      GSE201236</a
    >.
  </li>
  <li>
    RNA-Seq reads for Adipose are available at GEO accession
    <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE196572">
      GSE196572</a
    >.
  </li>
  <li>
    RNA-Seq reads for Liver are available at GEO accession
    <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE200453">
      GSE200453</a
    >.
  </li>
</ul>

<br />

<h2 id="studies"><a href="/download/study-data/">Original study data</a></h2>

<p>Archives of the data files used in the originating studies.</p>
