---
permalink: /download/
title: Download
layout: base
version: v3
---

{% assign version = page.version %} {% assign genome =
site.data.versions[version].genome %} {% assign tissues =
site.data.versions[version].tissues %}

<p>
  Download pages: <a href="/download/v1/">v1</a> •
  <a href="/download/v2/">v2</a> • v3 •
  <a href="/download/v3-unmerged/">v3 unmerged</a> •
  <a href="/download/study-data/">Study data</a>
</p>

<h1>Download data</h1>

<p>
  See the <a href="/about/#data-specs">About page</a> for processing, data
  format, and RatGTEx version number specifications. File names include the
  version number, and files that include genomic coordinates also include the
  genome build for clarity, even though only one genome build is used for each
  version.
</p>

<p><a href="/data/gene.{{ version }}.txt">Gene info</a></p>

<h2>Gene expression</h2>

<p>
  <a href="/data/medianGeneExpression.{{ version }}.txt.gz">
    Median TPM per gene per tissue
  </a>
  <em style="color: gray">Used for heatmap visualizations</em>
</p>

<strong>log2(readcount+1)</strong>
<em style="color: gray">Used to compute allelic fold change</em>
<p>
  {% for tissue in tissues %}
  <a
    href="/data/expr/expr.log2.{{ tissue.id }}.{{ version }}_{{ genome }}.bed.gz"
    title="{{ tissue.name }}"
  >
    {{ tissue.id }}
  </a>
  {% unless forloop.last %} • {% endunless %} {% endfor %}
</p>

<strong>TPM (transcripts per million)</strong>
<em style="color: gray">Relative abundance of transcripts</em>
<p>
  {% for tissue in tissues %}
  <a
    href="/data/expr/expr.tpm.{{ tissue.id }}.{{ version }}_{{ genome }}.bed.gz"
    title="{{ tissue.name }}"
  >
    {{ tissue.id }}
  </a>
  {% unless forloop.last %} • {% endunless %} {% endfor %}
</p>

<strong>Inverse-quantile normalized</strong>
<p>
  {% for tissue in tissues %}
  <a
    href="/data/expr/expr.iqn.{{ tissue.id }}.{{ version }}_{{ genome }}.bed.gz"
    title="{{ tissue.name }}"
  >
    {{ tissue.id }}
  </a>
  {% unless forloop.last %} • {% endunless %} {% endfor %}
</p>

<strong>Filtered inverse-quantile normalized</strong>
<em style="color: gray">Used for eQTL mapping</em>
<p>
  {% for tissue in tissues %}
  <a
    href="/data/expr/expr.iqn.filtered.{{ tissue.id }}.{{ version }}_{{ genome }}.bed.gz"
    title="{{ tissue.name }}"
  >
    {{ tissue.id }}
  </a>
  {% unless forloop.last %} • {% endunless %} {% endfor %}
</p>

<p>
  <strong><a href="/predict/">Predicted expression on all GWAS rats</a></strong>
</p>

<br />

<h2>Genotypes</h2>

<p>
  In this version, rats in the Adipose and Liver cohort are in the `round10_4`
  genotypes, and all others are in the `round10_5` genotypes. The included
  variants may differ between the two genotype files.
</p>

<ul>
  <li>
    <a href="/data/geno/ratgtex_v3_round10_4.{{ genome }}.vcf.gz">
      ratgtex_v3_round10_4.{{ genome }}.vcf.gz
    </a>
  </li>
  <li>
    <a href="/data/geno/ratgtex_v3_round10_5.{{ genome }}.vcf.gz">
      ratgtex_v3_round10_5.{{ genome }}.vcf.gz
    </a>
  </li>
</ul>

<br />

<h2>Other info</h2>

<p>
  <a href="/data/ref/RatGTEx_rats.{{ version }}.tsv">Rat info (all tissues)</a>
</p>

<p>
  <a href="/data/ref/RatGTEx_samples.{{ version }}.tsv">
    Sample info (all tissues)
  </a>
</p>

<strong>Rat IDs</strong>
<em style="color: gray">
  The final QC-filtered sample sets used for eQTL mapping
</em>
<br />
<p>
  {% for tissue in tissues %}
  <a
    href="/data/rat_ids/rat_ids.{{ tissue.id }}.{{ version }}.txt"
    title="{{ tissue.name }}"
  >
    {{ tissue.id }}
  </a>
  {% unless forloop.last %} • {% endunless %} {% endfor %}
</p>

<strong>RNA-Seq FASTQ file lists</strong>
<em style="color: gray">with corresponding rat IDs</em>
<br />
Each line includes one FASTQ path for single-end reads, or two FASTQ paths for
paired-end reads, followed by the rat ID. FASTQ maps can include a mix of
single-end and paired-end rows. These files may include mappings to samples that
were filtered out during QC, so use the rat ID lists above to get the final
sample sets.
<br />
<p>
  {% for tissue in tissues %}
  <a
    href="/data/fastq_map/fastq_map.{{ tissue.id }}.{{ version }}.txt"
    title="{{ tissue.name }}"
  >
    {{ tissue.id }}
  </a>
  {% unless forloop.last %} • {% endunless %} {% endfor %}
</p>

<strong>Covariates</strong>
<em style="color: gray">used for eQTL mapping</em>
<p>
  {% for tissue in tissues %}
  <a
    href="/data/covar/covar.{{ tissue.id }}.{{ version }}.txt"
    title="{{ tissue.name }}"
  >
    {{ tissue.id }}
  </a>
  {% unless forloop.last %} • {% endunless %} {% endfor %}
</p>

<br />

<h2>eQTLs</h2>

<p>
  <strong>
    <a href="/data/eqtl/eqtls_indep.{{ version }}_{{ genome }}.txt"
      >Conditionally independent cis-eQTLs</a
    >
  </strong>
  <br />
  A table of cis-eQTLs from all tissues.
</p>

<strong>All significant cis-eQTL variant-gene pairs</strong>

<p>
  {% for tissue in tissues %}
  <a
    href="/data/eqtl/cis_qtl_signif.{{ tissue.id }}.{{ version }}_{{ genome }}.txt.gz"
    title="{{ tissue.name }}"
  >
    {{ tissue.id }}
  </a>
  {% unless forloop.last %} • {% endunless %} {% endfor %}
</p>

<p>
  <strong>
    <a href="/data/eqtl/top_assoc.{{ version }}_{{ genome }}.txt">
      Top association per gene
    </a>
  </strong>
  <br />
  A table of the strongest variant-gene association per gene per tissue, even if
  not significant. A q-value threshold of 0.05 was used to determine significant
  eQTLs.
</p>

<strong>trans-eQTLs</strong>
<br />
All SNP-gene associations genome-wide with TSS distance &gt; 5 Mb and with an
arbitrary cutoff of p &lt; 1e-5. No significance testing was performed.
<p>
  {% for tissue in tissues %}
  <a
    href="/data/eqtl/trans_qtl_pairs.{{ tissue.id }}.{{ version }}_{{ genome }}.txt.gz"
    title="{{ tissue.name }}"
  >
    {{ tissue.id }}
  </a>
  {% unless forloop.last %} • {% endunless %} {% endfor %}
</p>

<br />

<h2>sQTLs</h2>

<strong>Splice phenotypes</strong>
<em style="color: gray">
  Relative usage of splice junctions within junction clusters
</em>
<p>
  {% for tissue in tissues %}
  <a
    href="/data/splice/leafcutter.{{ tissue.id }}.{{ version }}_{{ genome }}.bed.gz"
    title="{{ tissue.name }}"
  >
    {{ tissue.id }}
  </a>
  {% unless forloop.last %} • {% endunless %} {% endfor %}
</p>

<strong>Covariates</strong>
<em style="color: gray">used for sQTL mapping</em>
<p>
  {% for tissue in tissues %}
  <a
    href="/data/splice/covar_splice.{{ tissue.id }}.{{ version }}.txt"
    title="{{ tissue.name }}"
  >
    {{ tissue.id }}
  </a>
  {% unless forloop.last %} • {% endunless %} {% endfor %}
</p>

<p>
  <strong>
    <a href="/data/splice/sqtls_indep.{{ version }}_{{ genome }}.txt">
      Conditionally independent cis-sQTLs
    </a>
  </strong>
  <br />
  A table of cis-sQTLs from all tissues.
</p>

<strong>All significant cis-sQTL variant-phenotype pairs</strong>
<p>
  {% for tissue in tissues %}
  <a
    href="/data/splice/splice.cis_qtl_signif.{{ tissue.id }}.{{ version }}_{{ genome }}.txt.gz"
    title="{{ tissue.name }}"
  >
    {{ tissue.id }}
  </a>
  {% unless forloop.last %} • {% endunless %} {% endfor %}
</p>

<p>
  <strong>
    <a href="/data/splice/top_assoc_splice.{{ version }}_{{ genome }}.txt">
      Top association per gene
    </a>
  </strong>
  <br />
  A table of the strongest variant-gene association per gene per tissue, even if
  not significant. A q-value threshold of 0.05 was used to determine significant
  sQTLs.
</p>

<strong>trans-sQTLs</strong>
<br />
All SNP-phenotype associations genome-wide with TSS distance &gt; 5 Mb and with
an arbitrary cutoff of p &lt; 1e-5. No significance testing was performed.
<p>
  {% for tissue in tissues %}
  <a
    href="/data/splice/splice.trans_qtl_pairs.{{ tissue.id }}.{{ version }}_{{ genome }}.txt.gz"
    title="{{ tissue.name }}"
  >
    {{ tissue.id }}
  </a>
  {% unless forloop.last %} • {% endunless %} {% endfor %}
</p>

<br />

<h2>API</h2>

<p>
  Different types of data used for visualizations, including expression, eQTLs,
  and reference annotations, can be accessed via API. See the
  <a href="/about/api/">API documentation</a> for details.
</p>

<br />

<h2>Raw data</h2>

<ul>
  <li>
    RNA-Seq reads for the IL, LHb, NAcc, OFC, and PL brain regions are available
    in SRA and can be accessed from GEO accession
    <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE173141">
      GSE173141</a
    >.
  </li>
  <li>
    RNA-Seq reads for Eye are available at GEO accession
    <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE201236">
      GSE201236</a
    >.
  </li>
  <li>
    RNA-Seq reads for Adipose are available at GEO accession
    <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE196572">
      GSE196572</a
    >.
  </li>
  <li>
    RNA-Seq reads for Liver are available at GEO accession
    <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE200453">
      GSE200453</a
    >.
  </li>
</ul>

<br />

<h2 id="studies"><a href="/download/study-data/">Original study data</a></h2>

<p>Archives of the data files used in the originating studies.</p>
