<!doctype html>
<html lang="en">

<head>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1">

    <link rel="stylesheet" href="/assets/css/jquery-ui.css">
    <link href="https://cdn.jsdelivr.net/npm/bootstrap@5.0.1/dist/css/bootstrap.min.css" rel="stylesheet"
        integrity="sha384-+0n0xVW2eSR5OomGNYDnhzAbDsOXxcvSN1TPprVMTNDbiYZCxYbOOl7+AMvyTG2x" crossorigin="anonymous">
    <link rel="stylesheet" href="https://use.fontawesome.com/releases/v5.3.1/css/all.css"
        integrity="sha384-mzrmE5qonljUremFsqc01SB46JvROS7bZs3IO2EmfFsd15uHvIt+Y8vEf7N7fWAU" crossorigin="anonymous">
    <link rel="stylesheet" href="/assets/css/main.css">
    <link rel="icon" type="image/png" sizes="32x32" href="/assets/images/favicon-32x32.png">
    <link rel="icon" type="image/png" sizes="96x96" href="/assets/images/favicon-96x96.png">
    <link rel="icon" type="image/png" sizes="16x16" href="/assets/images/favicon-16x16.png">

    <script src="https://code.jquery.com/jquery-3.6.0.min.js"
        integrity="sha256-/xUj+3OJU5yExlq6GSYGSHk7tPXikynS7ogEvDej/m4=" crossorigin="anonymous"></script>
    <script src="https://code.jquery.com/ui/1.12.1/jquery-ui.min.js"
        integrity="sha256-VazP97ZCwtekAsvgPBSUwPFKdrwD3unUfSGVYrahUqU=" crossorigin="anonymous"></script>
    <script src="https://cdn.jsdelivr.net/npm/bootstrap@5.0.1/dist/js/bootstrap.bundle.min.js"
        integrity="sha384-gtEjrD/SeCtmISkJkNUaaKMoLD0//ElJ19smozuHV6z3Iehds+3Ulb9Bn9Plx0x4"
        crossorigin="anonymous"></script>
    <script src="/assets/js/main.js"></script>

    <title>About â€¢ RatGTEx</title>
</head>

<body>
    {% include navigation.html %}

    <div class="container">

        <h1 id="data-specs">Data Specifications</h1>

        <p>
            The datasets provided in this portal conform to the following specifications to facilitate comparisons across tissues.
            The raw data for the tissues in this portal come from multiple studies. The processed data, however, may differ from the results reported in each study for a couple reasons:
            <ol>
                <li>
                    Different processing decisions could be made for each study, while the data in this portal are processed uniformly to facilitate comparison across tissues.
                </li>
                <li>
                    The published study results are immutable, while new versions of data in this portal may be released, for example to improve quality or to use updated reference genome and annotations.
                </li>
            </ol>
            Nevertheless, we also host the original results from individual studies on the <a href="/download/#studies">Download</a> page when available.
        </p>

        <h4>Processing</h4>
        <p>
            The <a href="https://github.com/daniel-munro/ratgtex-pipeline">ratgtex-pipeline</a> repository contains all RNA-Seq and eQTL/sQTL mapping pipeline code. The steps are also on protocols.io: <a href="http://dx.doi.org/10.17504/protocols.io.rm7vzyk92lx1/v1">http://dx.doi.org/10.17504/protocols.io.rm7vzyk92lx1/v1</a>. The <a href="https://github.com/daniel-munro/ratgtex-server-data">ratgtex-server-data</a> repository contains code that processes those results into additional data files for download and for the API.
        </p>
        <ul>
            <li>Genome assembly and gene annotations used for each data release:
                <ul>
                    <li>v1:
                        <ul>
                            <li>Assembly: <a href="http://ftp.ensembl.org/pub/release-99/fasta/rattus_norvegicus/dna/Rattus_norvegicus.Rnor_6.0.dna.toplevel.fa.gz">Rnor_6.0 Ensembl</a> (GCA_000001895.4)</li>
                            <li>Annotations: <a href="http://ftp.ensembl.org/pub/release-99/gtf/rattus_norvegicus/Rattus_norvegicus.Rnor_6.0.99.gtf.gz">Ensembl Rat Release 99</a></li>
                        </ul>
                    </li>
                    <li>v2:
                        <ul>
                            <li>Assembly: <a href="http://ftp.ensembl.org/pub/release-108/fasta/rattus_norvegicus/dna/Rattus_norvegicus.mRatBN7.2.dna.toplevel.fa.gz">mRatBN7.2 Ensembl</a> (GCA_015227675.2)</li>
                            <li>Annotations: <a href="http://ftp.ensembl.org/pub/release-108/gtf/rattus_norvegicus/Rattus_norvegicus.mRatBN7.2.108.gtf.gz">Ensembl Rat Release 108</a></li>
                        </ul>
                    </li>
                    <li>v3:
                        <ul>
                            <li>Assembly: <a href="https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/015/227/675/GCF_015227675.2_mRatBN7.2/GCF_015227675.2_mRatBN7.2_genomic.fna.gz">mRatBN7.2 RefSeq</a> (GCA_015227675.2)</li>
                            <li>Annotations: <a href="https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/015/227/675/GCF_015227675.2_mRatBN7.2/GCF_015227675.2_mRatBN7.2_genomic.gtf.gz">mRatBN7.2 RefSeq</a></li>
                        </ul>
                    </li>
                </ul>
            </li>
            <li>RNA-Seq read counts, transformed as log<sub>2</sub>(count+1), were used to compute cis-eQTL effect sizes (allelic fold change).</li>
            <li>Inverse-quantile normalized expression values were used for eQTL mapping.</li>
            <li>eQTL mapping: <a href="https://github.com/broadinstitute/tensorqtl">tensorQTL</a>. cis-eQTL mapping tested SNPs in cis-windows +/- 1 Mb from each gene's transcription start site.</li>
            <li>sQTL mapping: splice junction usage was quantified using leafcutter, and sQTLs were mapped with tensorQTL, grouping phenotypes by gene. cis-sQTL mapping tested SNPs in cis-windows +/- 1 Mb from each gene's transcription start site.</li>
        </ul>

        <h4>Formats</h4>

        <h5>Gene expression <em style="color: gray;">(expr.{units}.{tissue}.{version}.bed.gz)</em></h5>
        <p>Expression tables are provided in BED format, with four columns describing the gene plus one column per sample. Only the transcription start sites are specified, not the whole gene length.</p>
        <ul>
            <li><strong>#chr</strong> Chromosome name, e.g. chr1</li>
            <li><strong>start</strong> Transcription start site. Coordinates in BED format are 0-based, so this is one less than the position as it is typically represented.</li>
            <li><strong>end</strong> One higher than the start value. Thus, this would be the typical coordinate to represent the TSS.</li>
            <li><strong>gene_id</strong> Gene symbol</li>
            <li><strong>{sample}</strong> Each additional column is labeled with the sample or rat ID and contains log2(count+1), TPM, or inverse-quantile normalized values.</li>
        </ul>

        <h5>All significant cis-eQTL gene-SNP pairs <em style="color: gray;">(cis_qtl_signif.{tissue}.{version}.txt.gz)</em></h5>
        <p>A compressed tab-separated table.</p>
        <ul>
            <li><strong>gene_id</strong> Gene symbol</li>
            <li><strong>variant_id</strong> SNP ID (e.g. chr1:669562)</li>
            <li><strong>tss_distance</strong> SNP position - TSS position, oriented on the gene's strand</li>
            <li><strong>af</strong> Alternative allele frequency</li>
            <li><strong>ma_samples</strong> Number of samples with minor allele</li>
            <li><strong>ma_count</strong> Total number of minor alleles (i.e. no. het. + 2 * no. hom. alt.)</li>
            <li><strong>pval_nominal</strong> Nominal p-value</li>
            <li><strong>slope</strong> Coefficient for the SNP genotype in the linear model</li>
            <li><strong>slope_se</strong> Standard error of the slope</li>
            <li><strong>pval_nominal_threshold</strong> Gene-tissue-specific nominal p-value significance threshold as determined by permutations and transcriptome-wide FDR</li>
        </ul>

        <h5>Strong associations from trans-eQTL mapping <em style="color: gray;">(trans_qtl_pairs.{tissue}.{version}.txt.gz)</em></h5>
        <p>All measured SNPs genome-wide were tested against expression of each gene, and pairs with p-value &lt; 1e-5 and &gt; 5 Mb TSS distance are included here. Rows are sorted by variant location.</p>
        <ul>
            <li><strong>variant_id</strong> SNP ID (e.g. chr1:669562)</li>
            <li><strong>gene_id</strong> Gene symbol</li>
            <li><strong>pval</strong> Nominal p-value</li>
            <li><strong>slope</strong> Coefficient (slope) for the SNP genotype in the linear model</li>
            <li><strong>slope_se</strong> Standard error of the slope</li>
            <li><strong>af</strong> Alternative allele frequency</li>
        </ul>

        <h5>Conditionally independent cis-eQTLs <em style="color: gray;">(eqtls_indep.{version}.txt)</em></h5>
        <p>A table of cis-eQTLs from all tissues. Stepwise regression was used to test for cis-eQTLs beyond and uncorrelated with the top cis-eQTL per gene.</p>
        <ul>
            <li><strong>tissue</strong> Tissue abbreviation</li>
            <li><strong>gene_id</strong> Gene symbol</li>
            <li><strong>num_var</strong> Number of cis-window variants tested for the specified gene in this tissue</li>
            <li><strong>variant_id</strong> SNP ID (e.g. chr1:669562) for the top eQTL SNP (eSNP). In cases of tied top SNPs that are in 100% LD, one was randomly chosen among them.</li>
            <li><strong>chrom</strong> Chromosome of the top SNP (e.g. chr1)</li>
            <li><strong>pos</strong> Position of the top SNP (bp)</li>
            <li><strong>ref</strong> Reference allele for the top SNP</li>
            <li><strong>alt</strong> Alternative allele for the top SNP</li>
            <li><strong>af</strong> Alternative allele frequency</li>
            <li><strong>tss_distance</strong> SNP position - TSS position, oriented on the gene's strand</li>
            <li><strong>pval_nominal</strong> Nominal p-value</li>
            <li><strong>slope</strong> Coefficient for the SNP genotype in the linear model</li>
            <li><strong>slope_se</strong> Standard error of the slope</li>
            <li><strong>pval_beta</strong> Empirical beta-approximated p-value</li>
            <li><strong>rank</strong> The strongest cis-eQTL found per gene-tissue has rank 1, next-strongest has rank 2, etc.</li>
            <li><strong>log2_aFC</strong> eQTL effect size measured as allelic fold change</li>
        </ul>

        <h5>Top association per gene <em style="color: gray;">(top_assoc.{version}.txt)</em></h5>
        <p>A table of the strongest variant-gene association per gene per tissue, even if not significant.</p>
        <ul>
            <li><strong>tissue</strong> Tissue abbreviation</li>
            <li><strong>gene_id</strong> Gene symbol</li>
            <li><strong>num_var</strong> Number of cis-window variants tested for the specified gene in this tissue</li>
            <li><strong>variant_id</strong> SNP ID (e.g. chr1:669562) for the top eQTL SNP (eSNP). In cases of tied top SNPs that are in 100% LD, one was randomly chosen among them.</li>
            <li><strong>chrom</strong> Chromosome of the top SNP (e.g. chr1)</li>
            <li><strong>pos</strong> Position of the top SNP (bp)</li>
            <li><strong>ref</strong> Reference allele for the top SNP</li>
            <li><strong>alt</strong> Alternative allele for the top SNP</li>
            <li><strong>af</strong> Alternative allele frequency</li>
            <li><strong>tss_distance</strong> SNP position - TSS position, oriented on the gene's strand</li>
            <li><strong>pval_nominal</strong> Nominal p-value</li>
            <li><strong>slope</strong> Coefficient for the SNP genotype in the linear model</li>
            <li><strong>slope_se</strong> Standard error of the slope</li>
            <li><strong>pval_beta</strong> Empirical beta-approximated p-value</li>
            <li><strong>qval</strong> Q-value used to control transcriptome-wide FDR per tissue (q &lt; 0.05 was used to determine significant cis-eQTLs)</li>
            <li><strong>pval_nominal_threshold</strong> Gene-tissue-specific nominal p-value significance threshold as determined by permutations and transcriptome-wide FDR</li>
            <li><strong>log2_aFC</strong> Effect size measured as allelic fold change</li>
        </ul>

        <!-- Splicing -->

        <h5>Splice phenotypes <em style="color: gray;">(leafcutter.{tissue}.{version}.bed.gz)</em></h5>
        <p>Splice phenotypes are provided in BED format, with four columns describing the splice junctions and genes plus one column per sample. Only the gene transcription start sites are specified, not the whole gene length or the splice junction itself, as these are used to determine the gene's cis-window for QTL mapping. See <a href="http://davidaknowles.github.io/leafcutter/">leafcutter docs</a> for descriptions of these junctions and junction clusters.</p>
        <ul>
            <li><strong>#Chr</strong> Chromosome name, e.g. chr1</li>
            <li><strong>start</strong> Transcription start site. Coordinates in BED format are 0-based, so this is one less than the position as it is typically represented.</li>
            <li><strong>end</strong> One higher than the start value. Thus, this would be the typical coordinate to represent the TSS.</li>
            <li><strong>ID</strong> Splice junction ID (chromosome:start:end:cluster_id:gene_id)</li>
            <li><strong>{sample}</strong> Each additional column is labeled with the sample or rat ID and contains inverse-quantile normalized values.</li>
        </ul>

        <h5>All significant cis-sQTL phenotype-SNP pairs <em style="color: gray;">(cis_qtl_signif.{tissue}.{version}.txt.gz)</em></h5>
        <p>A compressed tab-separated table.</p>
        <ul>
            <li><strong>phenotype_id</strong> Splice junction ID (chromosome:start:end:cluster_id:gene_id).</li>
            <li><strong>variant_id</strong> SNP ID (e.g. chr1:669562)</li>
            <li><strong>tss_distance</strong> SNP position - TSS position, oriented on the gene's strand</li>
            <li><strong>af</strong> Alternative allele frequency</li>
            <li><strong>ma_samples</strong> Number of samples with minor allele</li>
            <li><strong>ma_count</strong> Total number of minor alleles (i.e. no. het. + 2 * no. hom. alt.)</li>
            <li><strong>pval_nominal</strong> Nominal p-value</li>
            <li><strong>slope</strong> Coefficient for the SNP genotype in the linear model</li>
            <li><strong>slope_se</strong> Standard error of the slope</li>
            <li><strong>pval_nominal_threshold</strong> Gene-tissue-specific nominal p-value significance threshold as determined by permutations and transcriptome-wide FDR</li>
        </ul>

        <h5>Strong associations from trans-sQTL mapping <em style="color: gray;">(splice.trans_qtl_pairs.{tissue}.{version}.txt.gz)</em></h5>
        <p>All measured SNPs genome-wide were tested against each splice phenotype, and pairs with p-value &lt; 1e-5 and &gt; 5 Mb TSS distance are included here. Rows are sorted by variant location.</p>
        <ul>
            <li><strong>variant_id</strong> SNP ID (e.g. chr1:669562)</li>
            <li><strong>phenotype_id</strong> Splice junction ID (chromosome:start:end:cluster_id:gene_id).</li>
            <li><strong>pval</strong> Nominal p-value</li>
            <li><strong>slope</strong> Coefficient (slope) for the SNP genotype in the linear model</li>
            <li><strong>slope_se</strong> Standard error of the slope</li>
            <li><strong>af</strong> Alternative allele frequency</li>
        </ul>

        <h5>Conditionally independent cis-sQTLs <em style="color: gray;">(sqtls_indep.{version}.txt)</em></h5>
        <p>A table of cis-sQTLs from all tissues. Stepwise regression was used to test for cis-sQTLs beyond and uncorrelated with the top cis-sQTL per gene.</p>
        <ul>
            <li><strong>tissue</strong> Tissue abbreviation</li>
            <li><strong>phenotype_id</strong> Splice junction ID (chromosome:start:end:cluster_id:gene_id).</li>
            <li><strong>num_var</strong> Number of cis-window variants tested for the specified phenotype in this tissue</li>
            <li><strong>variant_id</strong> SNP ID (e.g. chr1:669562) for the top sQTL SNP (sSNP). In cases of tied top SNPs that are in 100% LD, one was randomly chosen among them.</li>
            <li><strong>chrom</strong> Chromosome of the top SNP (e.g. chr1)</li>
            <li><strong>pos</strong> Position of the top SNP (bp)</li>
            <li><strong>ref</strong> Reference allele for the top SNP</li>
            <li><strong>alt</strong> Alternative allele for the top SNP</li>
            <li><strong>af</strong> Alternative allele frequency</li>
            <li><strong>tss_distance</strong> SNP position - TSS position, oriented on the gene's strand</li>
            <li><strong>pval_nominal</strong> Nominal p-value</li>
            <li><strong>slope</strong> Coefficient for the SNP genotype in the linear model</li>
            <li><strong>slope_se</strong> Standard error of the slope</li>
            <li><strong>pval_beta</strong> Empirical beta-approximated p-value</li>
            <li><strong>gene_id</strong> Gene symbol, used as phenotype groups for sQTL mapping</li>
            <li><strong>group_size</strong> Number of splice phenotypes tested for the given gene</li>
            <li><strong>rank</strong> The strongest cis-sQTL found per gene-tissue has rank 1, next-strongest has rank 2, etc.</li>
        </ul>

        <h5>Top splice association per gene <em style="color: gray;">(top_assoc_splice.{version}.txt)</em></h5>
        <p>A table of the strongest variant-gene association per gene per tissue, even if not significant.</p>
        <ul>
            <li><strong>tissue</strong> Tissue abbreviation</li>
            <li><strong>phenotype_id</strong> Splice junction ID (chromosome:start:end:cluster_id:gene_id).</li>
            <li><strong>num_var</strong> Number of cis-window variants tested for the specified phenotype in this tissue</li>
            <li><strong>variant_id</strong> SNP ID (e.g. chr1:669562) for the top sQTL SNP (sSNP). In cases of tied top SNPs that are in 100% LD, one was randomly chosen among them.</li>
            <li><strong>chrom</strong> Chromosome of the top SNP (e.g. chr1)</li>
            <li><strong>pos</strong> Position of the top SNP (bp)</li>
            <li><strong>ref</strong> Reference allele for the top SNP</li>
            <li><strong>alt</strong> Alternative allele for the top SNP</li>
            <li><strong>af</strong> Alternative allele frequency</li>
            <li><strong>tss_distance</strong> SNP position - TSS position, oriented on the gene's strand</li>
            <li><strong>pval_nominal</strong> Nominal p-value</li>
            <li><strong>slope</strong> Coefficient for the SNP genotype in the linear model</li>
            <li><strong>slope_se</strong> Standard error of the slope</li>
            <li><strong>pval_beta</strong> Empirical beta-approximated p-value</li>
            <li><strong>gene_id</strong> Gene symbol, used as phenotype groups for sQTL mapping</li>
            <li><strong>group_size</strong> Number of splice phenotypes tested for the given gene</li>
            <li><strong>qval</strong> Q-value used to control transcriptome-wide FDR per tissue (q &lt; 0.05 was used to determine significant cis-sQTLs)</li>
            <li><strong>pval_nominal_threshold</strong> Gene-tissue-specific nominal p-value significance threshold as determined by permutations and transcriptome-wide FDR</li>
        </ul>

    </div>
</body>

</html>
