<!doctype html>
<html lang="en">

<head>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1">

    <link rel="stylesheet" href="/assets/css/jquery-ui.css">
    <link href="https://cdn.jsdelivr.net/npm/bootstrap@5.0.1/dist/css/bootstrap.min.css" rel="stylesheet"
        integrity="sha384-+0n0xVW2eSR5OomGNYDnhzAbDsOXxcvSN1TPprVMTNDbiYZCxYbOOl7+AMvyTG2x" crossorigin="anonymous">
    <link rel="stylesheet" href="https://use.fontawesome.com/releases/v5.3.1/css/all.css"
        integrity="sha384-mzrmE5qonljUremFsqc01SB46JvROS7bZs3IO2EmfFsd15uHvIt+Y8vEf7N7fWAU" crossorigin="anonymous">
    <link rel="stylesheet" href="/assets/css/main.css">
    <link rel="icon" type="image/png" sizes="32x32" href="/assets/images/favicon-32x32.png">
    <link rel="icon" type="image/png" sizes="96x96" href="/assets/images/favicon-96x96.png">
    <link rel="icon" type="image/png" sizes="16x16" href="/assets/images/favicon-16x16.png">

    <script src="https://code.jquery.com/jquery-3.6.0.min.js"
        integrity="sha256-/xUj+3OJU5yExlq6GSYGSHk7tPXikynS7ogEvDej/m4=" crossorigin="anonymous"></script>
    <script src="https://code.jquery.com/ui/1.12.1/jquery-ui.min.js"
        integrity="sha256-VazP97ZCwtekAsvgPBSUwPFKdrwD3unUfSGVYrahUqU=" crossorigin="anonymous"></script>
    <script src="https://cdn.jsdelivr.net/npm/bootstrap@5.0.1/dist/js/bootstrap.bundle.min.js"
        integrity="sha384-gtEjrD/SeCtmISkJkNUaaKMoLD0//ElJ19smozuHV6z3Iehds+3Ulb9Bn9Plx0x4"
        crossorigin="anonymous"></script>
    <script src="/assets/js/main.js"></script>


    <title>Download • RatGTEx</title>
</head>

<body>
    {% include navigation.html %}

    <div class="container">

        {% assign version = page.version %}
        {% assign genome = site.data.versions[version].genome %}
        {% assign tissues = site.data.versions[version].tissues %}

        <h1>Download data</h1>

        <p>See the <a href="/about/#data-specs">About page</a> for processing, data format, and RatGTEx version number specifications. File names include the version number, and files that include genomic coordinates also include the genome build for clarity, even though only one genome build is used for each version.</p>

        <p><a href="/data/gene.{{ version }}.txt">Gene info</a></p>

        <h2>Gene expression</h2>

        <p>
            <a href="/data/medianGeneExpression.{{ version }}.txt.gz">Median TPM per gene per tissue</a>
            <em style="color: gray;">Used for heatmap visualizations</em>
        </p>

        <strong>log2(readcount+1)</strong>
        <em style="color: gray;">Used to compute allelic fold change</em>
        <p>
            {% for tissue in tissues %}
            <a href="/data/expr/expr.log2.{{ tissue.id }}.{{ version }}_{{ genome }}.bed.gz" title="{{ tissue.name }}">{{ tissue.id }}</a>{% unless forloop.last %} • {% endunless %}
            {% endfor %}
        </p>

        <strong>TPM (transcripts per million)</strong>
        <em style="color: gray;">Relative abundance of transcripts</em>
        <p>
            {% for tissue in tissues %}
            <a href="/data/expr/expr.tpm.{{ tissue.id }}.{{ version }}_{{ genome }}.bed.gz" title="{{ tissue.name }}">{{ tissue.id }}</a>{% unless forloop.last %} • {% endunless %}
            {% endfor %}
        </p>

        <strong>Inverse-quantile normalized</strong>
        <em style="color: gray;">Used for eQTL mapping</em>
        <p>
            {% for tissue in tissues %}
            <a href="/data/expr/expr.iqn.{{ tissue.id }}.{{ version }}_{{ genome }}.bed.gz" title="{{ tissue.name }}">{{ tissue.id }}</a>{% unless forloop.last %} • {% endunless %}
            {% endfor %}
        </p>

        <br>

        <h2>Genotypes</h2>

        <p>
            Genotypes are grouped by the datasets from which they originated. The sample sets for the tissues within a dataset are overlapping subsets. Different datasets include different sets of SNPs in their genotypes.
        </p>

        <p>
            <ul>
                <li><a href="/data/geno/BLA_NAcc2_PL2.{{ genome }}.vcf.gz">BLA_NAcc2_PL2.{{ genome }}.vcf.gz</a></li>
                <li><a href="/data/geno/Eye.{{ genome }}.vcf.gz">Eye.{{ genome }}.vcf.gz</a></li>
                <li><a href="/data/geno/IL_LHb_NAcc_OFC_PL.{{ genome }}.vcf.gz">IL_LHb_NAcc_OFC_PL.{{ genome }}.vcf.gz</a></li>
                <li><a href="/data/geno/Adipose_Liver.{{ genome }}.vcf.gz">Adipose_Liver.{{ genome }}.vcf.gz</a></li>
                <li><a href="/data/geno/Brain.{{ genome }}.vcf.gz">Brain.{{ genome }}.vcf.gz</a></li>
            </ul>
        </p>
        
        <br>

        <h2>Other info</h2>

        <p><a href="/data/ref/RatGTEx_rats.{{ version }}.tsv">Rat info (all tissues)</a></p>

        <p><a href="/data/ref/RatGTEx_samples.{{ version }}.tsv">Sample info (all tissues)</a></p>

        <strong>Rat IDs</strong>
        <em style="color: gray;">The final QC-filtered sample sets used for eQTL mapping</em>
        <br>
        <p>
            {% for tissue in tissues %}
            <a href="/data/rat_ids/rat_ids.{{ tissue.id }}.{{ version }}.txt" title="{{ tissue.name }}">{{ tissue.id }}</a>{% unless forloop.last %} • {% endunless %}
            {% endfor %}
        </p>

        <strong>RNA-Seq FASTQ file lists</strong>
        <em style="color: gray;">with corresponding rat IDs</em>
        <br>
        Each line includes one FASTQ path for single-end reads, or two FASTQ paths for paired-end reads, followed by the rat ID. FASTQ maps can include a mix of single-end and paired-end rows. These files may include mappings to samples that were filtered out during QC, so use the rat ID lists above to get the final sample sets.
        <br>
        <p>
            {% for tissue in tissues %}
            <a href="/data/fastq_map/fastq_map.{{ tissue.id }}.{{ version }}.txt" title="{{ tissue.name }}">{{ tissue.id }}</a>{% unless forloop.last %} • {% endunless %}
            {% endfor %}
        </p>

        <strong>Covariates</strong>
        <em style="color: gray;">used for eQTL mapping</em>
        <p>
            {% for tissue in tissues %}
            <a href="/data/covar/covar.{{ tissue.id }}.{{ version }}.txt" title="{{ tissue.name }}">{{ tissue.id }}</a>{% unless forloop.last %} • {% endunless %}
            {% endfor %}
        </p>

        <br>

        <h2>eQTLs</h2>

        <p>
            <strong><a href="/data/eqtl/eqtls_indep.{{ version }}_{{ genome }}.txt">Conditionally independent cis-eQTLs</a></strong>
            <br>
            A table of cis-eQTLs from all tissues.
        </p>

        <strong>All significant cis-eQTL variant-gene pairs</strong>

        <p>
            {% for tissue in tissues %}
            <a href="/data/eqtl/cis_qtl_signif.{{ tissue.id }}.{{ version }}_{{ genome }}.txt.gz" title="{{ tissue.name }}">{{ tissue.id }}</a>{% unless forloop.last %} • {% endunless %}
            {% endfor %}
        </p>

        <p>
            <strong><a href="/data/eqtl/top_assoc.{{ version }}_{{ genome }}.txt">Top association per gene</a></strong>
            <br>
            A table of the strongest variant-gene association per gene per tissue, even if not significant. A
            q-value threshold of 0.05 was used to determine significant eQTLs.
        </p>

        <strong>trans-eQTLs</strong>
        <br>
        All SNP-gene associations genome-wide with TSS distance &gt; 5 Mb and with an arbitrary cutoff of p &lt; 1e-5. No significance testing was performed.
        <p>
            {% for tissue in tissues %}
            <a href="/data/eqtl/trans_qtl_pairs.{{ tissue.id }}.{{ version }}_{{ genome }}.txt.gz" title="{{ tissue.name }}">{{ tissue.id }}</a>{% unless forloop.last %} • {% endunless %}
            {% endfor %}
        </p>

        <br>

        <h2>sQTLs</h2>

        <strong>Splice phenotypes</strong>
        <em style="color: gray;">Relative usage of splice junctions within junction clusters</em>
        <p>
            {% for tissue in tissues %}
            <a href="/data/splice/leafcutter.{{ tissue.id }}.{{ version }}_{{ genome }}.bed.gz" title="{{ tissue.name }}">{{ tissue.id }}</a>{% unless forloop.last %} • {% endunless %}
            {% endfor %}
        </p>

        <strong>Covariates</strong>
        <em style="color: gray;">used for sQTL mapping</em>
        <p>
            {% for tissue in tissues %}
            <a href="/data/splice/covar_splice.{{ tissue.id }}.{{ version }}.txt" title="{{ tissue.name }}">{{ tissue.id }}</a>{% unless forloop.last %} • {% endunless %}
            {% endfor %}
        </p>

        <p>
            <strong><a href="/data/splice/sqtls_indep.{{ version }}_{{ genome }}.txt">Conditionally independent cis-sQTLs</a></strong>
            <br>
            A table of cis-sQTLs from all tissues.
        </p>

        <strong>All significant cis-sQTL variant-phenotype pairs</strong>
        <p>
            {% for tissue in tissues %}
            <a href="/data/splice/splice.cis_qtl_signif.{{ tissue.id }}.{{ version }}_{{ genome }}.txt.gz" title="{{ tissue.name }}">{{ tissue.id }}</a>{% unless forloop.last %} • {% endunless %}
            {% endfor %}
        </p>

        <p>
            <strong><a href="/data/splice/top_assoc_splice.{{ version }}_{{ genome }}.txt">Top association per gene</a></strong>
            <br>
            A table of the strongest variant-gene association per gene per tissue, even if not significant. A
            q-value threshold of 0.05 was used to determine significant sQTLs.
        </p>

        <strong>trans-sQTLs</strong>
        <br>
        All SNP-phenotype associations genome-wide with TSS distance &gt; 5 Mb and with an arbitrary cutoff of p &lt; 1e-5. No significance testing was performed.
        <p>
            {% for tissue in tissues %}
            <a href="/data/splice/splice.trans_qtl_pairs.{{ tissue.id }}.{{ version }}_{{ genome }}.txt.gz" title="{{ tissue.name }}">{{ tissue.id }}</a>{% unless forloop.last %} • {% endunless %}
            {% endfor %}
        </p>

        <br>

        <h2>API</h2>

        <p>
            Different types of data used for visualizations, including expression, eQTLs, and reference annotations, can
            be accessed via API. See the <a href="/about/api/">API documentation</a> for details.
        </p>

        <br>

        <h2>Raw data</h2>

        <ul>
            <li>RNA-Seq reads for the IL, LHb, NAcc, OFC, and PL brain regions are available in SRA and can be accessed from GEO accession <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE173141">GSE173141</a>.</li>
            <li>RNA-Seq reads for Eye are available at GEO accession <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE201236">GSE201236</a>.</li>
            <li>RNA-Seq reads for Adipose are available at GEO accession <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE196572">GSE196572</a>.</li>
            <li>RNA-Seq reads for Liver are available at GEO accession <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE200453">GSE200453</a>.</li>
        </ul>

        <br>

        <h2 id="studies"><a href="/download/study-data/">Original study data</a></h2>

        <p>
            Archives of the data files used in the originating studies.
        </p>

    </div>
</body>

</html>
