<!doctype html>
<html lang="en">

<head>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1">

    <link rel="stylesheet" href="/css/jquery-ui.css">
    <link href="https://cdn.jsdelivr.net/npm/bootstrap@5.0.1/dist/css/bootstrap.min.css" rel="stylesheet"
        integrity="sha384-+0n0xVW2eSR5OomGNYDnhzAbDsOXxcvSN1TPprVMTNDbiYZCxYbOOl7+AMvyTG2x" crossorigin="anonymous">
    <link rel="stylesheet" href="https://use.fontawesome.com/releases/v5.3.1/css/all.css"
        integrity="sha384-mzrmE5qonljUremFsqc01SB46JvROS7bZs3IO2EmfFsd15uHvIt+Y8vEf7N7fWAU" crossorigin="anonymous">
    <link rel="stylesheet" href="/css/main.css">
    <link rel="icon" type="image/png" sizes="32x32" href="/images/favicon-32x32.png">
    <link rel="icon" type="image/png" sizes="96x96" href="/images/favicon-96x96.png">
    <link rel="icon" type="image/png" sizes="16x16" href="/images/favicon-16x16.png">

    <script src="https://code.jquery.com/jquery-3.6.0.min.js"
        integrity="sha256-/xUj+3OJU5yExlq6GSYGSHk7tPXikynS7ogEvDej/m4=" crossorigin="anonymous"></script>
    <script src="https://code.jquery.com/ui/1.12.1/jquery-ui.min.js"
        integrity="sha256-VazP97ZCwtekAsvgPBSUwPFKdrwD3unUfSGVYrahUqU=" crossorigin="anonymous"></script>
    <script src="https://cdn.jsdelivr.net/npm/bootstrap@5.0.1/dist/js/bootstrap.bundle.min.js"
        integrity="sha384-gtEjrD/SeCtmISkJkNUaaKMoLD0//ElJ19smozuHV6z3Iehds+3Ulb9Bn9Plx0x4"
        crossorigin="anonymous"></script>
    <script src="/src/main.js"></script>


    <title>Download | RatGTEx</title>
</head>

<body>
    <nav class="navbar navbar-expand-lg navbar-light bg-light">
        <div class="container-fluid">
            <!-- <a class="navbar-brand" href="/">RatGTEx Portal</a> -->
            <a class="navbar-brand" href="/">
                <img src="/images/RatGTExPortal.png" width="200px" alt="RatGTEx Portal">
            </a>
            <button class="navbar-toggler" type="button" data-bs-toggle="collapse"
                data-bs-target="#navbarSupportedContent" aria-controls="navbarSupportedContent" aria-expanded="false"
                aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="navbarSupportedContent">
                <ul class="navbar-nav me-auto mb-2 mb-lg-0">
                    <li class="nav-item">
                        <a class="nav-link" href="/">Home</a>
                    </li>
                    <li class="nav-item">
                        <a class="nav-link active" aria-current="page" href="/download/">Download</a>
                    </li>
                    <li class="nav-item dropdown">
                        <a class="nav-link dropdown-toggle" href="#" id="browseDropdown" role="button"
                            data-bs-toggle="dropdown" aria-expanded="false">Browse</a>
                        <ul class="dropdown-menu" aria-labelledby="browseDropdown">
                            <li>
                                <h6 class="dropdown-header">Expression</h6>
                            </li>
                            <li><a class="dropdown-item" href="/top-expressed/">Top expressed genes</a></li>
                            <li><a class="dropdown-item" href="/query-gene-expression/">Query gene expression</a></li>
                            <li>
                                <hr class="dropdown-divider">
                            </li>
                            <li>
                                <h6 class="dropdown-header">eQTLs</h6>
                            </li>
                            <li><a class="dropdown-item" href="/eqtl-dashboard/">eQTL violin plots</a></li>
                            <li><a class="dropdown-item" href="/gene-eqtl-visualizer/">Gene eQTL Visualizer</a></li>
                        </ul>
                    </li>
                    <li class="nav-item dropdown">
                        <a class="nav-link dropdown-toggle" href="#" id="aboutDropdown" role="button"
                            data-bs-toggle="dropdown" aria-expanded="false">About</a>
                        <ul class="dropdown-menu" aria-labelledby="aboutDropdown">
                            <li><a class="dropdown-item" href="/about/">RatGTEx</a></li>
                            <li><a class="dropdown-item" href="/about/samples/">Samples</a></li>
                            <li><a class="dropdown-item" href="/about/api/">API</a></li>
                        </ul>
                    </li>                </ul>
                <form class="d-flex col-md-4 mb-0" id="gene-search">
                    <div class="input-group">
                        <input class="form-control basicAutoComplete" id="search-input" type="text" autocomplete="off"
                            placeholder="Gene Search (symbol or Ensembl ID)">
                        <button class="btn btn-outline-secondary" type="submit">
                            <i class="fa fa-search"></i>
                        </button>
                    </div>
                </form>
            </div>
        </div>
    </nav>

    <div class="container">
        <h1>Download data</h1>

        <!-- <p><a href="/data/ref/metadata.csv">Sample metadata</a></p> -->

        <p>See the <a href="/about/">About page</a> for processing and data format specifications.</p>

        <p><a href="/data/gene.txt">Gene info</a></p>

        <h2>Raw data</h2>

        <p>RNA-Seq reads for the IL, LHb, NAcc, OFC, and PL brain regions are available in SRA and can be accessed from GEO accession <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE173141">GSE173141</a>.</p>

        <h2>Gene expression</h2>

        <strong>log2(count+1)</strong>
        <em style="color: gray;">Used to compute allelic fold change</em>
        <p>
            <a href="/data/expr/BLA.expr.log2.bed.gz">BLA</a> |
            <a href="/data/expr/Eye.expr.log2.bed.gz">Eye</a> |
            <a href="/data/expr/IL.expr.log2.bed.gz">IL</a> |
            <a href="/data/expr/LHb.expr.log2.bed.gz">LHb</a> |
            <a href="/data/expr/NAcc.expr.log2.bed.gz">NAcc</a> |
            <a href="/data/expr/NAcc2.expr.log2.bed.gz">NAcc2</a> |
            <a href="/data/expr/OFC.expr.log2.bed.gz">OFC</a> |
            <a href="/data/expr/PL.expr.log2.bed.gz">PL</a> |
            <a href="/data/expr/PL2.expr.log2.bed.gz">PL2</a>
        </p>

        <strong>TPM (transcripts per million)</strong>
        <em style="color: gray;">Relative abundance of transcripts</em>
        <p>
            <a href="/data/expr/BLA.expr.tpm.bed.gz">BLA</a> |
            <a href="/data/expr/Eye.expr.tpm.bed.gz">Eye</a> |
            <a href="/data/expr/IL.expr.tpm.bed.gz">IL</a> |
            <a href="/data/expr/LHb.expr.tpm.bed.gz">LHb</a> |
            <a href="/data/expr/NAcc.expr.tpm.bed.gz">NAcc</a> |
            <a href="/data/expr/NAcc2.expr.tpm.bed.gz">NAcc2</a> |
            <a href="/data/expr/OFC.expr.tpm.bed.gz">OFC</a> |
            <a href="/data/expr/PL.expr.tpm.bed.gz">PL</a> |
            <a href="/data/expr/PL2.expr.tpm.bed.gz">PL2</a>
        </p>

        <strong>Inverse-quantile normalized</strong>
        <em style="color: gray;">Used for eQTL mapping</em>
        <p>
            <a href="/data/expr/BLA.expr.iqn.bed.gz">BLA</a> |
            <a href="/data/expr/Eye.expr.iqn.bed.gz">Eye</a> |
            <a href="/data/expr/IL.expr.iqn.bed.gz">IL</a> |
            <a href="/data/expr/LHb.expr.iqn.bed.gz">LHb</a> |
            <a href="/data/expr/NAcc.expr.iqn.bed.gz">NAcc</a> |
            <a href="/data/expr/NAcc2.expr.iqn.bed.gz">NAcc2</a> |
            <a href="/data/expr/OFC.expr.iqn.bed.gz">OFC</a> |
            <a href="/data/expr/PL.expr.iqn.bed.gz">PL</a> |
            <a href="/data/expr/PL2.expr.iqn.bed.gz">PL2</a>
        </p>

        <h2>Genotypes</h2>

        <p>
            Genotypes are grouped by the datasets from which they originated. The sample sets for the tissues within a dataset are overlapping subsets. Different datasets include different sets of SNPs in their genotypes.
        </p>

        <p>
            <!-- <table>
                <thead><tr><th>Genotype file</th><th>Tissues</th></tr></thead>
                <tr><td><a href="/data/geno/eye.vcf.gz">eye.vcf.gz</a></td><td>Eye</td></tr>
                <tr><td><a href="/data/geno/brain_regions.vcf.gz">brain_regions.vcf.gz</a></td><td>IL, LHb, NAcc, OFC, PL</td></tr>
            </table> -->
            <ul>
                <li><a href="/data/geno/BLA_NAcc2_PL2.vcf.gz">BLA_NAcc2_PL2.vcf.gz</a></li>
                <li><a href="/data/geno/Eye.vcf.gz">Eye.vcf.gz</a></li>
                <li><a href="/data/geno/IL_LHb_NAcc_OFC_PL.vcf.gz">IL_LHb_NAcc_OFC_PL.vcf.gz</a></li>
            </ul>
        </p>
        
        <h2>eQTL covariates</h2>

        <p>
            <a href="/data/covar/BLA.covar.txt">BLA</a> |
            <a href="/data/covar/Eye.covar.txt">Eye</a> |
            <a href="/data/covar/IL.covar.txt">IL</a> |
            <a href="/data/covar/LHb.covar.txt">LHb</a> |
            <a href="/data/covar/NAcc.covar.txt">NAcc</a> |
            <a href="/data/covar/NAcc2.covar.txt">NAcc2</a> |
            <a href="/data/covar/OFC.covar.txt">OFC</a> |
            <a href="/data/covar/PL.covar.txt">PL</a> |
            <a href="/data/covar/PL2.covar.txt">PL2</a>
        </p>

        <h2>eQTLs</h2>

        <p>
            <strong><a href="/data/eqtl/eqtls_indep.txt">Conditionally independent cis-eQTLs</a></strong>
            <br>
            A table of cis-eQTLs from all tissues.
        </p>

        <strong>All significant cis-eQTL variant-gene pairs</strong>

        <p>
            <a href="/data/eqtl/BLA.cis_qtl_signif.txt.gz">BLA</a> |
            <a href="/data/eqtl/Eye.cis_qtl_signif.txt.gz">Eye</a> |
            <a href="/data/eqtl/IL.cis_qtl_signif.txt.gz">IL</a> |
            <a href="/data/eqtl/LHb.cis_qtl_signif.txt.gz">LHb</a> |
            <a href="/data/eqtl/NAcc.cis_qtl_signif.txt.gz">NAcc</a> |
            <a href="/data/eqtl/NAcc2.cis_qtl_signif.txt.gz">NAcc2</a> |
            <a href="/data/eqtl/OFC.cis_qtl_signif.txt.gz">OFC</a> |
            <a href="/data/eqtl/PL.cis_qtl_signif.txt.gz">PL</a> |
            <a href="/data/eqtl/PL2.cis_qtl_signif.txt.gz">PL2</a>
        </p>

        <p>
            <strong><a href="/data/eqtl/top_assoc.txt">Top association per gene</a></strong>
            <br>
            A table of the strongest variant-gene association per gene per tissue, even if not significant. A
            q-value threshold of 0.05 was used to determine significant eQTLs.
        </p>

        <strong>trans-eQTLs</strong>
        <br>
        All SNP-gene associations genome-wide with TSS distance &gt; 5 Mb and with an arbitrary cutoff of p &lt; 1e-5
        <p>
            <a href="/data/eqtl/BLA.trans_qtl_pairs.txt.gz">BLA</a> |
            <a href="/data/eqtl/Eye.trans_qtl_pairs.txt.gz">Eye</a> |
            <a href="/data/eqtl/IL.trans_qtl_pairs.txt.gz">IL</a> |
            <a href="/data/eqtl/LHb.trans_qtl_pairs.txt.gz">LHb</a> |
            <a href="/data/eqtl/NAcc.trans_qtl_pairs.txt.gz">NAcc</a> |
            <a href="/data/eqtl/NAcc2.trans_qtl_pairs.txt.gz">NAcc2</a> |
            <a href="/data/eqtl/OFC.trans_qtl_pairs.txt.gz">OFC</a> |
            <a href="/data/eqtl/PL.trans_qtl_pairs.txt.gz">PL</a> |
            <a href="/data/eqtl/PL2.trans_qtl_pairs.txt.gz">PL2</a>
        </p>

        <!-- <h2>sQTLs</h2>

        <p>
            Splice QTLs are currently available for only IL, LHb, NAcc, OFC, and PL tissues.
        </p>

        <p>
            <strong><a href="/data/sqtl/sqtls_indep.txt">Conditionally independent cis-sQTLs</a></strong>
            <br>
            A table of cis-sQTLs from all tissues.
        </p>

        <p>
            <strong><a href="/data/sqtl/top_assoc_splice.txt">Top splice association per gene</a></strong>
            <br>
            A table of the strongest variant-gene splice association per intron cluster per tissue, even if not significant. A
            q-value threshold of 0.05 was used to determine significant sQTLs, grouped by gene for the purposes of multiple testing.
        </p> -->

        <h2>API</h2>

        <p>
            Different types of data used for visualizations, including expression, eQTLs, and reference annotations, can
            be accessed via API. See the <a href="/about/api/">API documentation</a> for details.
        </p>

        <h2 id="studies">Original study data</h2>

        <p>
            The processed data above may differ from the results reported in each original study for a couple reasons:
            <ol>
                <li>
                    Different processing decisions could be made for each study, while the data in this portal are processed uniformly to facilitate comparison across tissues.
                </li>
                <li>
                    The published study results are immutable, while new versions of data in this portal may be released, for example to improve quality or to use updated reference genome and annotations.
                </li>
            </ol>
            Therefore, we also provide data files here corresponding directly to the results in each paper.
        </p>

        <h5>IL, LHb, NAcc, OFC, PL</h5>
        <p>
            <a href="https://www.biorxiv.org/content/10.1101/2022.04.07.487560v1">The regulatory landscape of multiple brain regions in outbred heterogeneous stock rats</a>
        </p>
        <ul>
            <li><a href="data/studies/IL_LHb_NAcc_OFC_PL/metadata_p50_hao_chen_2014.csv">metadata_p50_hao_chen_2014.csv</a></li>
            <li><a href="data/studies/IL_LHb_NAcc_OFC_PL/ensembl-gene_raw-counts.txt">ensembl-gene_raw-counts.txt</a></li>
            <li><a href="data/studies/IL_LHb_NAcc_OFC_PL/ensembl-gene_raw-tpm.txt">ensembl-gene_raw-tpm.txt</a></li>
            <li><a href="data/studies/IL_LHb_NAcc_OFC_PL/ensembl-gene_inv-quant_PL.bed.gz">ensembl-gene_inv-quant_PL.bed.gz</a></li>
            <li><a href="data/studies/IL_LHb_NAcc_OFC_PL/ensembl-gene_log2_IL.bed.gz">ensembl-gene_log2_IL.bed.gz</a></li>
            <li><a href="data/studies/IL_LHb_NAcc_OFC_PL/ensembl-gene_log2_LHb.bed.gz">ensembl-gene_log2_LHb.bed.gz</a></li>
            <li><a href="data/studies/IL_LHb_NAcc_OFC_PL/ensembl-gene_log2_NAcc.bed.gz">ensembl-gene_log2_NAcc.bed.gz</a></li>
            <li><a href="data/studies/IL_LHb_NAcc_OFC_PL/ensembl-gene_log2_OFC.bed.gz">ensembl-gene_log2_OFC.bed.gz</a></li>
            <li><a href="data/studies/IL_LHb_NAcc_OFC_PL/ensembl-gene_log2_PL.bed.gz">ensembl-gene_log2_PL.bed.gz</a></li>
            <li><a href="data/studies/IL_LHb_NAcc_OFC_PL/ensembl-gene_inv-quant_IL.bed.gz">ensembl-gene_inv-quant_IL.bed.gz</a></li>
            <li><a href="data/studies/IL_LHb_NAcc_OFC_PL/ensembl-gene_inv-quant_LHb.bed.gz">ensembl-gene_inv-quant_LHb.bed.gz</a></li>
            <li><a href="data/studies/IL_LHb_NAcc_OFC_PL/ensembl-gene_inv-quant_NAcc.bed.gz">ensembl-gene_inv-quant_NAcc.bed.gz</a></li>
            <li><a href="data/studies/IL_LHb_NAcc_OFC_PL/ensembl-gene_inv-quant_OFC.bed.gz">ensembl-gene_inv-quant_OFC.bed.gz</a></li>
            <li><a href="data/studies/IL_LHb_NAcc_OFC_PL/top_assoc.txt">top_assoc.txt</a></li>
            <li><a href="data/studies/IL_LHb_NAcc_OFC_PL/eqtls_indep.txt">eqtls_index.txt</a></li>
            <li><a href="data/studies/IL_LHb_NAcc_OFC_PL/ASE_aFC.txt">ASE_aFC.txt</a></li>
            <li><a href="data/studies/IL_LHb_NAcc_OFC_PL/top_assoc_splice.txt">top_assoc_splice.txt</a></li>
            <li><a href="data/studies/IL_LHb_NAcc_OFC_PL/sqtls_indep.txt">sqtls_indep.txt</a></li>
        </ul>

    </div>
</body>

</html>
